src/vm-memory/src/bitmap.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+//! Atomic bitmap implementation.
+
+// Temporarly disable unused warnings.
+// TODO: remove these once the Bitmap integration is completed.
+#![allow(dead_code)]
+
+use std::sync::atomic::{AtomicU64, Ordering};
+
+/// `Bitmap` implements a simple bit map on the page level with test and set operations. It is
+/// page-size aware, so it converts addresses to page numbers before setting or clearing the bits.
+#[derive(Debug)]
+pub struct Bitmap {
+    map: Vec<AtomicU64>,
+    size: usize,
+    page_size: usize,
+}
+
+impl Bitmap {
+    /// Create a new bitmap of `byte_size`, with one bit per `page_size`.
+    /// In reality this is rounded up, and you get a new vector of the next multiple of 64 bigger
+    /// than `size` for free.
+    pub fn new(byte_size: usize, page_size: usize) -> Self {
+        // Bit size is the number of bits in the bitmap, always at least 1 (to store the state of
+        // the '0' address).
+        let bit_size = std::cmp::max(1, byte_size / page_size);
```

Comment:
nit:
```suggestion
        let num_pages = std::cmp::max(1, byte_size / page_size);
```
for clarity.